### DISCUSSION TITLE
disable joint stereo

### DISCUSSION THREAD

**Discussion by cybercuate** [AzuraCast/AzuraCast#7638] [General]
is there any way to use stereo 2 channels instead of joint stereo for mp3s mount points? foobar shows all my mp3 streams are joint stereo, only FLAC is 2 channels. Maybe I can do something with this line? "output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128)"

**Comment by LordHelmchen666** [comment]
From LS Documentation

> stereo_mode: One of: "stereo", "joint_stereo" or "default" (default: "default"). Default means that the underlying library (libmp3lame) will pick the stereo mode based on compression ration and input channels.

So it would be `output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128, stereo_mode="stereo")`
You could use https://github.com/RM-FM/ls-config-replace to insert this.

**Reply by Marcioghiraldelli** [reply]
I could NOT quite understand how this works, wouldn't it be just a matter of inserting this line in LS_CONFIG last? In the last field after the default one? I tested it and it didn't give any errors but I don't know if it worked.

**Reply by LordHelmchen666** [reply]
I am not sure it's ok just to "overwrite" in LS_config. Maybe @gAlleb would know this.
But anyway i searched joint_stereo vs. stereo usage with mp3. And all results in found recommending to use joint_stereo for Bitrate lower than 256kbps. So maybe you should keep it as it is?

**Reply by gAlleb** [reply]
By default you have `radio.mp3` mountpoint. To use custom settings you don't need to overwrite config. You can tweak it with AzuraCast native functionality:
1. Go to  `Broadcasting` - `Edit Liquidsoap Configuration`. Remember your `output.icecast` settings: password and so on.
2. Go to `Broadcasting` - `Mount Points` 
3. Pick your mount point (e.g. `radio.mp3`) and in `Edit Mount Point` window choose the `AutoDJ` tab - then disable AutoDJ by clicking the slider. Save.
4. Go to `Broadcasting` - `Edit Liquidsoap Configuration` and add mount point manually (which you have had before) before `# Local Broadcasts` or after (somewhere there :)) But add `stereo_mode="stereo"` to it. Save. 
5. That's it.

Sometimes there's a lot of redundancy between channels. `Joint stereo` takes this redundancy into account and instead of encoding channels left and right, it encodes middle and side. Well, sometimes it sounds better, sometimes not. Usually it uses available bits more efficiently. So choice is yours.

Apart from using `%mp3(samplerate=44100, stereo=true, bitrate=128, stereo_mode="stereo")` you can do `internal_quality=0`
`%mp3(samplerate=44100, stereo=true, bitrate=128, stereo_mode="stereo", internal_quality=0)`

which will give you the `highest quality` encoding. By default it's not the `highest`.

**Reply by cybercuate** [reply]
so these are my new settings
**for CBR320**
output.icecast(%mp3.cbr(samplerate=44100, stereo=true, bitrate=320, stereo_mode="stereo", id3v2=true, internal_quality=0)
and for **CBR128**
output.icecast(%mp3(samplerate=44100, stereo=true, bitrate=128, stereo_mode="stereo", _internal_quality=4_)
and now is working fine, foobar shows the stereo_mode is stereo. Now, liquidsoap's internal quality value by default is 2 but after reading this:
[https://hydrogenaud.io/index.php/topic,125216.0.html](https://hydrogenaud.io/index.php/topic,125216.0.html)
when using Lame3.100 (or anything above 3.97), which is installed by liquidsoap, at 160kbps or less  -q4 produces better sound quality than -q3 and above (-q3 to -q0)

Thanks for the help and I'd love to see these settings on azuracast's UI so we can change them easily
